content/chapters/software-stack/lab/content/basic-syscall.md

```diff
@@ -3,6 +3,27 @@
 To get a better grasp on how the software stack works, let's do a bottom-up approach:
 we build and run different programs, that start of by using the system call API (the lowest layer in the software stack) and progressively use higher layers.
 
+### System Calls Explained
+
+A system call, or syscall for short, is a method used by applications to communicate with the operating system's kernel.
+
+The need for syscalls is tied to the modern operating systems model of conceptually separating into kernel space and user space.
+
+The kernel space manages the hardware resources such as CPU, I/O devices, disk or memory.
+Moreover, the kernel also provides an interface for the user space applications to interact with the hardware.
+
+The user space is where you are running your applications and processes.
+From the user space, we cannot directly access the hardware or perform privileged operations.
+You need to use syscalls to perform privileged operations such as accessing the hardware.
+
+Below, you can see some examples of system calls and what resource they request from the kernel:
+
+- `brk()` is used to allocate memory
+- `open()` is used to access the file system and open a specific file
+- `write()` is used to access the file system and modify the contents of a specific file
```

Comment:
```suggestion
- `brk()` is used to allocate memory

- `open()` is used to access the file system and open a specific file

- `write()` is used to access the file system and modify the contents of a specific file
```
IDK why markdownlint doesn't complain about this, but it should. If you have lists that span multiple sentences, they're difficult to read unless you leave some blank lines between list items.

Reply:
It only complains about heterogeneous lists, e.g.: tests that have `+` and `-`.  Though, I think the default line spacing in lists is enough

Reply:
It seems to me a bit too much to add these newlines. But it is not a strong opinion. I changed it as requested.